Implement Offerings

xTh3Unkn0wnx opened the pull request:
![Screenshot 2024-06-24 141550](https://github.com/acm-ucr/aiatucr/assets/97139185/2c9dbb92-f31e-42af-8d1b-3cf3a7070bec)

- Created a component for Offering Cards
- Created a second component to render multiple Offering cards and hold the title for what's being offered.
- Added the OfferingCards component onto the landing page (page.js) 
- Closes #6

khandrew1 commented on src/app/page.js:
```diff
@@ -1,5 +1,11 @@
 const Home = () => {
-  return <>Hello World</>;
+  return (
+    <>
```
Please remove `<>`

khandrew1 commented on src/components/landing/OfferingCard.js:
```diff
@@ -0,0 +1,25 @@
+import { FaUserAlt } from "react-icons/fa";
+
+const OfferingCard = ({ title, description, icon }) => {
+  if (!icon && !title && !description) {
+    return <div></div>;
+  }
+  if (!title) {
+    title = "Title";
+  }
+  if (!icon) {
+    icon = <FaUserAlt className="size-12 text-[#4F9AD7]" />;
+  }
+  if (!description) {
+    description = "Description";
+  }
```
If statements are unnecessary, as this is a static website, none of these props should be or ever will be blank

khandrew1 commented on src/components/landing/OfferingCard.js:
```diff
@@ -0,0 +1,25 @@
+    description = "Description";
+  }
+  return (
+    <div className="h-80 w-72 rounded-3xl p-8 bg-[#AEC6D8] flex flex-col justify-between">
```
Don't use raw hex codes in the color, should be added as a custom color in the config

khandrew1 commented on src/components/landing/OfferingCard.js:
```diff
@@ -0,0 +1,25 @@
+    description = "Description";
+  }
+  return (
+    <div className="h-80 w-72 rounded-3xl p-8 bg-[#AEC6D8] flex flex-col justify-between">
```
Also don't use specific values for the height, keep it at percentage values (fractions)

khandrew1 replied:
vh is fine too but only use when necessary

khandrew1 commented on src/components/landing/OfferingCards.js:
```diff
@@ -0,0 +1,62 @@
+      "Engaging sessions with AI experts, industry professionals, and prominent researchers.",
+    icon: <RiPresentationFill className="size-12 text-[#4F9AD7]" />,
+  },
+  null,
```
remove these nulls, i know the design was weird but may look better without them

khandrew1 commented on src/components/landing/OfferingCards.js:
```diff
@@ -0,0 +1,62 @@
+  return (
+    <section className="p-4">
+      <h2
+        className="font-[Montserrat
```
again don't worry about the font

khandrew1 commented on src/components/landing/OfferingCards.js:
```diff
@@ -0,0 +1,62 @@
+          {Offerings.map((offering, index) => (
+            <OfferingCard
+              key={index}
+              title={offering?.title}
```
question marks aren't needed

khandrew1 reviewed (changes requested):
feel free to ask about any questions

xTh3Unkn0wnx commented:
![Screenshot 2024-06-24 212901](https://github.com/acm-ucr/aiatucr/assets/97139185/ee07cecc-6acf-4089-86ed-bd1c487bf5f3)

- Removed the react fragment (<>...</>) from the page.js
- Changed the height parameter for the card to a percentage
- Removed question marks when mapping the cards in OfferingCards.js
- Removed the font-[xxx] class, will worry about that later when we go over it
- Removed the nulls in the arrays of cards
- Changed the classes so it does not use [xxx] in the Classname
- Added two custom colors (The text gray and the blue for events bg and 404)
- Added the custom classes to OfferingCard.js so it can be used.
- Closed #6

xTh3Unkn0wnx commented:
If needed, I can use some random blue and gray color for the background and text color in their respective elements if we do not want to deal with tailwind configs just yet.

khandrew1 commented on tailwind.config.js:
```diff
@@ -9,6 +9,10 @@ module.exports = {
   theme: {
     extend: {
       ai: {},
```
put the colors in here, again like the other acm projects
example:
`ai : { "blue": "#AEC6D8" }`

khandrew1 commented on tailwind.config.js:
```diff
@@ -9,6 +9,10 @@ module.exports = {
   theme: {
     extend: {
       ai: {},
+      colors: {
+        "ai-event-blue": "#AEC6D8",
+        "ai-text-gray": "#2E2E2E",
+      },
```
remove this

khandrew1 commented on src/components/landing/OfferingCards.js:
```diff
@@ -0,0 +1,54 @@
+    title: "AI NEWS AND OPPORTUNITIES",
+    description:
+      "Stay informed with the most recent developments, breakthroughs, and job opportunities in the AI realm.",
+    icon: <LiaNewspaperSolid className="size-12 text-[#4F9AD7]" />,
```
don't use size for the icons, use font sizes like `text-xl` or whatever fits

khandrew1 replied:
also color, don't use raw hex codes.

khandrew1 commented on src/components/landing/OfferingCard.js:
```diff
@@ -0,0 +1,11 @@
+const OfferingCard = ({ title, description, icon }) => {
+  return (
+    <div className="h-full w-72 rounded-3xl p-8 bg-ai-event-blue flex flex-col justify-between">
```
widths should also not use pixel values

khandrew1 commented on src/components/landing/OfferingCards.js:
```diff
@@ -0,0 +1,54 @@
+  },
+];
+
+const OfferingCards = ({}) => {
```
remove empty object, if there are no props parameters should stay empty, `()`

khandrew1 commented on src/components/landing/OfferingCards.js:
```diff
@@ -0,0 +1,54 @@
+const OfferingCards = ({}) => {
+  return (
+    <section className="p-4">
+      <h2 className="text-4xl text-[#4F9AD7] text-center m-8">We Offer...</h2>
```
do not use `<h2>` tags (it can mess with SEO)
instead use `<p>` tags with font sizes given in the tailwind docs

xTh3Unkn0wnx commented:
![Screenshot 2024-06-27 213043](https://github.com/acm-ucr/aiatucr-website/assets/97139185/67b63a40-ae01-4aff-a389-e01ad318a85d)

- Changed the tailwind config to match the desired format and scheme mentioned by lead
- changed width to use fractional/percentage units instead
- removed an empty object for the OfferingCards component 
- Changed an element from h2 to p 
- Used text-5xl for icons rather than size-x
- Closed #6
